cmd/influx_inspect/export/export.go

```diff
@@ -0,0 +1,395 @@
+		if files, ok := cmd.tsmFiles[key]; ok {
+			fmt.Printf("writing out tsm file data for %s...", key)
+			if err := cmd.writeTsmFiles(w, files); err != nil {
+				return err
+			}
+			fmt.Println("complete.")
+		}
+		if _, ok := cmd.walFiles[key]; ok {
+			fmt.Printf("writing out wal file data for %s...", key)
+			if err := cmd.writeWALFiles(w, cmd.walFiles[key]); err != nil {
+				return err
+			}
+			fmt.Println("complete.")
+		}
+	}
+	return nil
+}
+
+func (cmd *Command) writeTsmFiles(w io.WriteCloser, files []string) error {
+	fmt.Fprintln(w, "# writing tsm data")
+
+	// use a function here to close the files in the defers and not let them accumulate in the loop
+	write := func(f string) error {
+		file, err := os.OpenFile(f, os.O_RDONLY, 0600)
+		if err != nil {
+			return fmt.Errorf("%v", err)
+		}
+		defer file.Close()
+		reader, err := tsm1.NewTSMReader(file)
+		if err != nil {
+			log.Printf("unable to read %s, skipping\n", f)
+			return nil
+		}
+		defer reader.Close()
+
+		if sgStart, sgEnd := reader.TimeRange(); sgStart > cmd.endTime || sgEnd < cmd.startTime {
+			return nil
+		}
+
+		for i := 0; i < reader.KeyCount(); i++ {
+			var pairs string
+			key, typ := reader.KeyAt(i)
+			values, _ := reader.ReadAll(string(key))
+			measurement, field := tsm1.SeriesAndFieldFromCompositeKey(key)
+
+			for _, value := range values {
+				if (value.UnixNano() < cmd.startTime) || (value.UnixNano() > cmd.endTime) {
+					continue
+				}
+
+				switch typ {
+				case tsm1.BlockFloat64:
+					pairs = field + "=" + fmt.Sprintf("%v", value.Value())
+				case tsm1.BlockInteger:
+					pairs = field + "=" + fmt.Sprintf("%vi", value.Value())
+				case tsm1.BlockBoolean:
+					pairs = field + "=" + fmt.Sprintf("%v", value.Value())
+				case tsm1.BlockString:
+					pairs = field + "=" + fmt.Sprintf("%q", models.EscapeStringField(fmt.Sprintf("%s", value.Value())))
+				default:
+					pairs = field + "=" + fmt.Sprintf("%v", value.Value())
+				}
+
+				fmt.Fprintln(w, string(measurement), pairs, value.UnixNano())
+			}
+		}
+		return nil
+	}
+
+	for _, f := range files {
+		if err := write(f); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (cmd *Command) writeWALFiles(w io.WriteCloser, files []string) error {
+	fmt.Fprintln(w, "# writing wal data")
+
+	// we need to make sure we write the same order that the wal received the data
+	sort.Strings(files)
+
+	// use a function here to close the files in the defers and not let them accumulate in the loop
+	write := func(f string) error {
+		file, err := os.OpenFile(f, os.O_RDONLY, 0600)
+		if err != nil {
+			return fmt.Errorf("%v", err)
+		}
+		defer file.Close()
+
+		reader := tsm1.NewWALSegmentReader(file)
+		defer reader.Close()
+		for reader.Next() {
+			entry, err := reader.Read()
+			if err != nil {
+				n := reader.Count()
+				fmt.Fprintf(os.Stderr, "file %s corrupt at position %d", file.Name(), n)
+				break
+			}
+
+			switch t := entry.(type) {
+			case *tsm1.WriteWALEntry:
```

Comment:
Yes, those entries.  It might be helpful to load the measurement/tags that are deleted at least once.  The values in that entries is the measurement+tags+field so you could strip out the field and just load the measurement+tags at least once.